### PR TITLE
[feat] add lang-elixir workspace chunk and image

### PR DESCRIPTION
**Pull request by emattiza**
## Description
Create new `workspace-elixir` image for elixir based gitpod projects. This is to add a new chunk, and make a standalone `workspace-elixir` image based off `workspace-base`. There is no intent to add this to `workspace-full`
## Related Issue(s)
Implements #775 

## How to test
Tests included in tests/lang-elixir.yaml (version output testing, latest elixir from https://www.erlang-solutions.com/downloads/)
## Release Notes
```
add workspace-elixir image
```
## Documentation
This adds a standard elixir and erlang toolchain workspace to the available gitpod workspace images for quicker bootstrapping of existing projects.

### CODE REVIEW

**Comment by utam0k** [comment]
@emattiza Hi, first of all, thank you for your first contribution! Is it possible to specify the version of erlang to be installed and test for that. Such as [golang](https://github.com/gitpod-io/workspace-images/tree/main/chunks/lang-go).

**Comment by emattiza** [comment]
> @emattiza Hi, first of all, thank you for your first contribution! Is it possible to specify the version of erlang to be installed and test for that. Such as [golang](https://github.com/gitpod-io/workspace-images/tree/main/chunks/lang-go).

It sure is. I can put in a commit for this add it as a configuration option, much like it is for golang.

File: chunks/lang-elixir/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+ARG base
+FROM ${base}
+
+USER root
+
+# Dazzle does not rebuild a layer until one of its lines are changed. Increase this counter to rebuild this layer.
+ENV TRIGGER_REBUILD=1
+
+RUN wget https://packages.erlang-solutions.com/erlang-solutions_2.0_all.deb \
+    && sudo dpkg -i erlang-solutions_2.0_all.deb \
```
Comment (princerachit) [inline chunks/lang-elixir/Dockerfile:10]: do we need this `sudo`? User is already set to root on line 4

File: chunks/lang-elixir/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    && apt update \
+    && install-packages \
+        esl-erlang \
+    && install-packages \
+        elixir
+
+
```
Comment (princerachit) [inline chunks/lang-elixir/Dockerfile:17]: let us get rid of one extra line.

File: chunks/lang-elixir/Dockerfile (hunk 1 above)
Comment (emattiza) [reply chunks/lang-elixir/Dockerfile:10]: agreed, will push changes

File: chunks/lang-elixir/Dockerfile (hunk 2 above)
Comment (emattiza) [reply chunks/lang-elixir/Dockerfile:17]: agreed, can do both in single step

File: tests/lang-elixir.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+- desc: it should run elixir
+  command: [elixir,--version]
+  assert:
+  - status == 0
```
Comment (utam0k) [inline tests/lang-elixir.yaml:4]: Could you please change this test so that we can check the version of elixir?

Comment (emattiza) [reply tests/lang-elixir.yaml:4]: loading this morning and will test build

File: chunks/lang-elixir/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,15 @@
+ARG base
+FROM ${base}
+
+USER root
+
+# Dazzle does not rebuild a layer until one of its lines are changed. Increase this counter to rebuild this layer.
+ENV TRIGGER_REBUILD=1
+
+RUN curl -fsSL https://packages.erlang-solutions.com/ubuntu/erlang_solutions.asc | sudo gpg --dearmor -o /usr/share/keyrings/erlang_solutions.gpg.key \
```
Comment (emattiza) [inline chunks/lang-elixir/Dockerfile:9]: This now follows the gpg dearmor guidance from the CONTRIBUTING.md guide

File: chunks/lang-elixir/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    && echo "deb [signed-by=/usr/share/keyrings/erlang_solutions.gpg.key] https://packages.erlang-solutions.com/ubuntu focal contrib" | sudo tee /etc/apt/sources.list.d/erlang.list > /dev/null \
+    && apt update \
+    && install-packages \
```
Comment (emattiza) [inline chunks/lang-elixir/Dockerfile:12]: erlang now installs as a dependency

File: tests/lang-elixir.yaml [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+  - stdout.indexOf("Erlang/OTP 24") != -1 ||
```
Comment (emattiza) [inline tests/lang-elixir.yaml:5]: this our test logic for the version. defaults to this when unversioned, we can look at specific versions in the repo and using a chunk.yaml if needed

**Review by utam0k — APPROVED** [review]
LGTM

**Review by princerachit — CHANGES_REQUESTED** [review]
Hey @emattiza ,
Thanks for the PR. Can you look at the [contribution guide and update other files](https://github.com/gitpod-io/workspace-images/blob/main/CONTRIBUTING.md#updating-release-configuration-files) which are required to publish an image?

Also Please update the description of the PR so it matches the default PR template.

--------



## Description


## Related Issue(s)

Fixes #

## How to test


## Release Notes

```release-note
```

## Documentation

**Comment by emattiza** [comment]
Heres what I show as todo:
- [x] Clean up PR description
- [x] Add chunk [link here](https://github.com/gitpod-io/workspace-images/blob/main/CONTRIBUTING.md#adding-a-chunk)
- [x] Add tests [link here](https://github.com/gitpod-io/workspace-images/blob/main/CONTRIBUTING.md#writing-tests)
- [x] Update root dazzle.yaml [link here](https://github.com/gitpod-io/workspace-images/blob/main/CONTRIBUTING.md#updating-dazzleyml)
- [x] Update publish workflows [link here](https://github.com/gitpod-io/workspace-images/blob/main/CONTRIBUTING.md#updating-release-configuration-files)

Anything I missed?

**Review by princerachit — APPROVED** [review]

**Pull request merged**